src/core/gltf2importer/bufferaccessorparser.cpp

```diff
@@ -135,6 +195,86 @@ bool Kuesa::GLTF2Import::BufferAccessorParser::parse(const QJsonArray &bufferAcc
         accessor.max = jsonArrayToVectorOfFloats(json.value(KEY_MAX).toArray());
         accessor.min = jsonArrayToVectorOfFloats(json.value(KEY_MIN).toArray());
         accessor.name = json.value(KEY_NAME).toString();
+
+        if (json.contains(KEY_SPARSE)) {
+            QJsonObject sparse = json.value(KEY_SPARSE).toObject();
+            if (!sparse.contains(KEY_COUNT) || !sparse.contains(KEY_SPARSE_INDICES) || !sparse.contains(KEY_SPARSE_VALUES)) {
+                qCWarning(kuesa, "Missing sparse data in accessor");
+                return false;
+            }
+
+            accessor.sparseCount = sparse.value(KEY_COUNT).toInt(0);
+
+            QJsonObject sparseIndices = sparse.value(KEY_SPARSE_INDICES).toObject();
+            accessor.sparseIndices.type = accessorTypeFromJSON(sparseIndices.value(KEY_COMPONENTTYPE).toInt(GL_UNSIGNED_INT));
+            accessor.sparseIndices.bufferViewIndex = sparseIndices.value(KEY_BUFFERVIEW).toInt(-1);
+            accessor.sparseIndices.offset = sparseIndices.value(KEY_BYTEOFFSET).toInt(0);
+
+            QJsonObject sparseValues = sparse.value(KEY_SPARSE_VALUES).toObject();
+            accessor.sparseValues.bufferViewIndex = sparseValues.value(KEY_BUFFERVIEW).toInt(-1);
+            accessor.sparseValues.offset = sparseValues.value(KEY_BYTEOFFSET).toInt(0);
+
+            if (accessor.sparseIndices.bufferViewIndex == -1 || accessor.sparseValues.bufferViewIndex == -1) {
+                qCWarning(kuesa, "Missing buffers for sparse indices or values in accessor");
+                return false;
+            }
+
+            QByteArray sparseIndicesData = context->bufferView(accessor.sparseIndices.bufferViewIndex).bufferData;
+            QByteArray sparseValuesData = context->bufferView(accessor.sparseValues.bufferViewIndex).bufferData;
+            if (sparseIndicesData.isEmpty() || sparseValuesData.isEmpty()) {
+                qCWarning(kuesa, "Empty buffers for sparse indices or values in accessor");
+                return false;
+            }
+
+            int stride = 0;
+            if (accessor.bufferData.isEmpty())
+                accessor.bufferData = QByteArray(accessor.dataSize * accessor.count, 0);
+            else {
+                accessor.bufferData.detach();
```

Comment:
I'm not sure I understand why there's a detach here.

Reply:
we're going to modify the values in the bufferView, but the original content might be used in some other accessor. Sparse values are an optimisation to only store (in the gltf file) the delta between the original and modified content. At runtime you need both states